### ISSUE TITLE
hiddenSettings applied through policy.json in Firefox are not read by uBlock Origin

### ISSUE THREAD

**Issue by emanruse** [uBlockOrigin/uBlock-issues#3095]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

`hiddenSettings` applied through `policy.json` in Firefox are not read by uBlock Origin.

### A specific URL where the issue occurs.

```yaml
irrelevant
```

### Steps to Reproduce

1. Create a `/usr/lib64/firefox/distribution/policies.json` file containing:

```
{
  "policies": {
    "3rdparty": {
      "Extensions": {
	"uBlock0@raymondhill.net": {
	  "adminSettings": {
	    "userSettings": {
	      "advancedUserEnabled": true
	    },
	    "hiddenSettings": {
	      "blockingProfiles": "11111/#F00 11011/#C0F 11001/#00F 11111"
	    }
	  }
	}
      }
    }
  }
}
```

2. Install uBlock Origin
(Optionally, also restart Firefox - not necessary but I tried that too)

3. Visit the Dashboard

### Expected behavior

The values inside the `hiddenSettings` section in the policy should be read by uBO, just like other settings.

### Actual behavior

The values inside the `hiddenSettings` section in the policy are **not** read by uBO. All other settings work fine, e.g. in the above simplified example, `advancedUserEnabled` is read correctly, however visiting the "Advanced settings" menu shows the default value (ending with 00001, not with 11111 as in policy):

```
blockingProfiles 11111/#F00 11010/#C0F 11001/#00F 00001
```

and the line is not bolded (as modified ones normally appear).

Testing with other hiddenSettings results in the same - those specific values in the policy are ignored by uBO.

#### Note

`about:policies` shows the correct values of everything, as set in `policies.json`.

### uBO version

1.55.0

### Browser name and version

Firefox 121.0.1 (64-bit), Tor Browser 13.0.9

### Operating System and version

Fedora 38, Whonix 17 (Debian 12 derivative)

**Comment by gorhill** [comment]
Works fine on my side. I do not have a `/usr/lib64/firefox` on my side, but it works when creating the policies in `/etc/firefox/policies/policies.json`.

**Comment by gwarser** [comment]
https://github.com/uBlockOrigin/uBlock-issues/issues/2795 ?

**Comment by gorhill** [comment]
Oh I see, read wrong and misunderstood. Right, same issue.

**Comment by gorhill** [comment]
@emanruse 

Use:

```json
{
  "policies": {
    "3rdparty": {
      "Extensions": {
        "uBlock0@raymondhill.net": {
          "userSettings": [
            [
              "advancedUserEnabled",
              true
            ]
          ],
          "advancedSettings": [
            [
              "blockingProfiles",
              "11111/#F00 11011/#C0F 11001/#00F 11111"
            ]
          ]
        }
      }
    }
  }
}
```

I changed the documentation to give prominence to this better way of preconfiguring uBO. The `adminSettings` is still supported but I do not intent to fix whatever is not working with it.

**Comment by gorhill** [comment]
Sorry my bad, this should be:

```json
          "userSettings": [
            [
              "advancedUserEnabled",
              "true"
            ]
          ],
```
Instead of:

```json
          "userSettings": [
            [
              "advancedUserEnabled",
              true
            ]
          ],
```

**Comment by gorhill** [comment]
Managed settings are defined by [their schema](https://github.com/gorhill/uBlock/blob/master/platform/common/managed_storage.json), it defines which settings has which type of value (string, boolean, etc). Last thing I want is a huge and hard to maintain schema file for every single settings, which I would have to modify every time I change the content of user settings. The most flexible way is a array of pairs of string `[ key, value ]`, this means I don't have to mind `managed_storage.json` each time I modify anything that touches user settings.

The justification is that it's just the way I chose things to be since I am the one doing the labor work to maintain all this -- so I chose what require no labor to maintain.

If something doesn't work as expected now that I have detailed how this work, just share details -- as said it's quite simple, each setting is a pair of string values inside an array, i.e. `[ "advancedUserEnabled", "true" ]` -- uBO does the casting internally to the proper value type from the string.
